sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -47,13 +47,14 @@
 SFP_STATUS_REMOVED = '0'
 SFP_STATUS_INSERTED = '1'
 
-# SFP error code enum, new elements can be added to the enum if new errors need to be supported.
-SFP_STATUS_ERR_ENUM = Enum('SFP_STATUS_ERR_ENUM', ['SFP_STATUS_ERR_I2C_STUCK', 'SFP_STATUS_ERR_BAD_EEPROM',
-                                                   'SFP_STATUS_ERR_UNSUPPORTED_CABLE', 'SFP_STATUS_ERR_HIGH_TEMP',
-                                                   'SFP_STATUS_ERR_BAD_CABLE'], start=2)
-
-# Convert the error code to string and store them in a set for convenience
-errors_block_eeprom_reading = set(str(error_code.value) for error_code in SFP_STATUS_ERR_ENUM)
+# SFP error code dictinary, new elements can be added if new errors need to be supported.
+SFP_STATUS_ERR_DICT = {
+    2: 'SFP_STATUS_ERR_I2C_STUCK',
+    4: 'SFP_STATUS_ERR_BAD_EEPROM',
+    8: 'SFP_STATUS_ERR_UNSUPPORTED_CABLE',
+    16: 'SFP_STATUS_ERR_HIGH_TEMP',
+    32: 'SFP_STATUS_ERR_BAD_CABLE'
+}
```

Comment:
maybe we still need to keep a set of "errors_block_eeprom_reading" as a subset of the whole errors, meaning not all errors will block the EEPROM reading, only on some certain error we will stop reading the EEPROM, but on others we will still continue to read.

Reply:
According to previous code implementation, all kinds of errors will be treated as "blocking eeprom read error". I suppose there will be no new errors in short term, could we keep it this way for now?

Reply:
@keboliu please let me know if you still have concern about this.